src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs

```diff
@@ -67,33 +67,30 @@ private static MethodInfo FirstImplementsIEquatableOfSecond(Type first, Type sec
             {
                 if (implementation.Argument.IsAssignableFrom(second))
                 {
-                    if (mostDerived.Argument == null)
+                    if (mostDerived.Argument == null || mostDerived.Argument.IsAssignableFrom(implementation.Argument))
                         mostDerived = implementation;
-                    else if (mostDerived.Argument.IsAssignableFrom(implementation.Argument))
-                    {
-                        // Look for method with most-derived of type arg which can be assigned to 'second'
-                        // If there end up being multiple of same type, look for most derived declaring type
-                        if (mostDerived.Argument != implementation.Argument
-                            || mostDerived.Argument.IsAssignableFrom(implementation.Argument))
-                        mostDerived = implementation;
-                    }
                 }
             }
 
             return mostDerived.Method;
         }
 
-        private static IEnumerable<EquatableMethodImpl> GetEquatableImplementations(Type type)
+        private static EquatableMethodImpl[] GetEquatableImplementations(Type type)
         {
-            foreach (Type @interface in type.GetInterfaces())
+            static bool IsIEquatableOfT(Type t, object filter) => t.IsGenericType && t.GetGenericTypeDefinition().Equals(typeof(IEquatable<>));
+            
+            var interfaces = type.FindInterfaces((t,f) => IsIEquatableOfT(t,f), string.Empty);
+            var implementations = new EquatableMethodImpl[interfaces.Length];
+
+            for(var i = 0; i < interfaces.Length; i++)
             {
-                if (@interface.GetTypeInfo().IsGenericType && @interface.GetGenericTypeDefinition().Equals(typeof(IEquatable<>)))
-                {
-                    var iMap = type.GetInterfaceMap(@interface);
-                    foreach (var method in iMap.TargetMethods)
-                        yield return new EquatableMethodImpl(method, method.GetParameters()[0].ParameterType, iMap.TargetType);
-                }
+                var iMap = type.GetInterfaceMap(interfaces[i]);
+                var method = iMap.TargetMethods[0];
```

Comment:
This is incorrect.  It should be:
```
var method = iMap.InterfaceMethods[0];
```

Reply:
Noting the followup at https://github.com/nunit/nunit/issues/3683#issuecomment-739479852:

> It appears that the problem is in the third part implementation of InterfaceMap I was using.
> 
> You may remember that all of my work is on WindowsCE and the Compact Framework.
> 
> Since the Compact Framework does not support InterfaceMap, I had to use a third party implementation of it.
> 
> Apparently, that implementation is faulty.
> 
> I am writing my own implementation, which should solve my problem.
> 
> Sorry for the noise here.